draft-js-drag-n-drop-upload-plugin/src/handleDroppedFiles.js

```diff
@@ -26,84 +28,61 @@ export default function onDropFile(config) {
-        //     if (blocks.size) {
-        //       const newEditorStateOrBlockType = handleBlock
-        //         ? handleBlock(newEditorState, newEditorState.getSelection(), file)
-        //         : defaultBlockType;
-        //
-        //       newEditorState = replaceBlock(
-        //         modifyBlockData(
-        //           newEditorState,
-        //           blocks.first().get('key'),
-        //           retainSrc ? { progress: undefined } : { progress: undefined, src: undefined }
-        //         ),
-        //         blocks.first().get('key'),
-        //         newEditorStateOrBlockType
-        //       );
-        //     } /* else {
-        //       const newEditorStateOrBlockType = handleBlock
-        //         ? handleBlock(newEditorState, newEditorState.getSelection(), file)
-        //         : defaultHandleBlock(newEditorState, newEditorState.getSelection(), file, defaultBlockType);
-        //
-        //       if (!newEditorStateOrBlockType) {
-        //         newEditorState = defaultHandleBlock(newEditorState, selection, file, defaultBlockType);
-        //       } else if (typeof newEditorStateOrBlockType === 'string') {
-        //         newEditorState = defaultHandleBlock(newEditorState, selection, file, newEditorStateOrBlockType);
-        //       } else {
-        //         newEditorState = newEditorStateOrBlockType;
-        //       }
-        //     } */
-        //   });
-        //
-        //   // Propagate progress
-        //   if (handleProgress) handleProgress(null);
-        //   setEditorState(newEditorState);
-        // }, () => {
-        //   // console.error(err);
-        // }, (percent) => {
-        //   // On progress, set entity data's progress field
-        //   let newEditorState = getEditorState();
-        //   placeholders.forEach((placeholder) => {
-        //     const blocks = getBlocksWhereEntityData(newEditorState, (p) => p.src === placeholder.src && p.progress !== undefined);
-        //     if (blocks.size) {
-        //       newEditorState = modifyBlockData(newEditorState, blocks.first().get('key'), { progress: percent });
-        //     }
-        //   });
-        //   setEditorState(newEditorState);
-        //
-        //   // Propagate progress
-        //   if (handleProgress) {
-        //     handleProgress(percent);
-        //   }
-        // });
+        handleUpload(data, (uploadedFiles) => {
+          const editorStateWithImages = uploadedFiles.reduce((editorState, file) => {
+            const placeholderBlock = getPlaceholderBlock(editorState, file);
+
+            if (!placeholderBlock) {
+              return editorState;
+            }
+
+            return handleBlock(editorState, placeholderBlock, file);
+          }, getEditorState());
+          setEditorState(editorStateWithImages);
+        }, (err) => {
+          console.error(err);
```

Comment:
It'd be appropriate to use a callback here, which you'd supply to the plugin config.